lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+                       "must be masked")
+            raise ValueError(message)
+        masked_weights_cube = cube.copy(data=weights_data)
+        masked_weights_cube.rename("weights")
+        return masked_weights_cube
+
+    def create_fuzzy_spatial_weights(self, masked_weights_cube):
+        """
+        Create fuzzy weights around points in the masked_weights_cube with
+        zero weight.
+
+        Args:
+            masked_weights_cube (iris.cube.Cube):
+                A cube containing an initial set of weights based on the mask
+                on the input cube.
+        Returns:
+            result (iris.cube.Cube):
+                A cube contiaing the fuzzy weights calculated based on the
+                masked_weights_cube. The dimension order may have changed from
+                the input cube as it has been sliced over x and y coordinates.
+        """
+        result = iris.cube.CubeList()
+        x_coord = masked_weights_cube.coord(axis='x').name()
+        y_coord = masked_weights_cube.coord(axis='y').name()
+        # The distance_transform_edt works on N-D cubes, so we want to make
+        # sure we only apply it to x-y slices.
+        for weights in masked_weights_cube.slices([x_coord, y_coord]):
+            if np.all(weights.data == 1.0):
+                # distance_transform_edt doesn't produce what we want if there
+                # are no zeros present.
+                result.append(weights.copy())
+            else:
+                fuzzy_data = distance_transform_edt(weights.data == 1., 1)
+                fuzzy_data = fuzzy_data.astype(np.float32)
+                rescaled_fuzzy_data = rescale(
+                    fuzzy_data, data_range=[0., self.fuzzy_length],
+                    clip=True)
+                result.append(weights.copy(data=rescaled_fuzzy_data))
+        result = result.merge_cube()
+        return result
+
+    @staticmethod
+    def multiply_weights(masked_weights_cube, one_dimensional_weights_cube,
+                         collapsing_coord):
+        """
+        Multiply two cubes together by taking slices along the coordinate
+        matching the collapsing_coord string.
+
+        Args:
+            masked_weights_cube (iris.cube.Cube):
+                A cube with spatial weights and any other leading dimensions.
+                This cube must have a coordinate matching the name given by
+                collapsing_coord and this coordinate must be the same length
+                as the corresponding coordinate in the
+                one_dimensional_weights_cube.
+            one_dimensional_weights_cube (iris.cube.Cube):
+                A cube with one_dimensional weights. The only dimension
+                coordinate in this cube matches the string given by
+                collapsing_coord and the length of this coord must match the
+                length of the same coordinate in masked_weights_cube.
+            collapsing_coord (string):
```

Comment:
Given the terminology used in other blending and weights plugins, could we call this eg `blend_coord`?